dev/tests/js/jasmine/tests/app/code/Magento/Customer/frontend/js/customer-data.test.js

```diff
@@ -95,84 +97,100 @@ define([
     }
 
     describe('Magento_Customer/js/customer-data', function () {
+
         beforeAll(function () {
             clearLocalStorage();
         });
 
-        beforeEach(function () {
-            jQueryGetJSON = $.getJSON;
+        beforeEach(function (done) {
+            originalGetJSON = jQuery.getJSON;
             sectionConfig['Magento_Customer/js/section-config'](sectionConfigSettings);
+
+            injector.require([
+                'Magento_Customer/js/customer-data'
+            ], function (Constr) {
+                obj = Constr;
+                done();
+            });
         });
 
         afterEach(function () {
-            $.getJSON = jQueryGetJSON;
-            clearLocalStorage();
+            try {
+                jQuery.getJSON = originalGetJSON;
+                clearLocalStorage();
+                injector.clean();
+                injector.remove();
+            } catch (e) {
+            }
```

Comment:
Are we expecring to have any exceptions here? I don't think it's good to have empty catch, it will just mask any issues inside

Reply:
Squire throws errors itself, this happens with anything that's using Squire same as in section-config.test.js

TypeError: undefined is not an object (evaluating 'getContext(this.id).defined')

Reply:
At most I can remove the try catch, thoughts?

Reply:
Let's keep it as is for now